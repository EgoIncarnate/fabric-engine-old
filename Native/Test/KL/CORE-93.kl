struct Vec3
{
  Scalar x;
  Scalar y;
  Scalar z;
};

struct Quad 
{ 
  Vec3 normal; 
}; 

function Vec3( Scalar x, Scalar y, Scalar z )
{
  self.x = x;
  self.y = y;
  self.z = z;
}

function Quad( in Vec3 v[] ) 
{ 
  self.normal = v[0]; 
} 

function entry() 
{ 
  var Vec3 v[]; 
  v.push( Vec3( 1.0, 2.0, 3.0 ) );
   
  var Quad q( v ); 

  report( q ); 
} 
