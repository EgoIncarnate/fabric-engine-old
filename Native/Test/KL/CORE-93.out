GlobalList
  StructDecl( 'Vec3' )
    StructDeclMember( 'x', Scalar )
    StructDeclMember( 'y', Scalar )
    StructDeclMember( 'z', Scalar )
  StructDecl( 'Quad' )
    StructDeclMember( 'normal', Vec3 )
  Function( '' ('__construct_Vec3__Scalar__Scalar__Scalar'), void )
    ParamList
      Param( 'self', Vec3, l-value )
      Param( 'x', Scalar, r-value )
      Param( 'y', Scalar, r-value )
      Param( 'z', Scalar, r-value )
    CompoundStatement
      StatementList
        ExprStatement
          AssignOp(ASSIGN_OP)
            StructMemberOp( 'x' )
              Var('self')
            Var('x')
        ExprStatement
          AssignOp(ASSIGN_OP)
            StructMemberOp( 'y' )
              Var('self')
            Var('y')
        ExprStatement
          AssignOp(ASSIGN_OP)
            StructMemberOp( 'z' )
              Var('self')
            Var('z')
  Function( '' ('__construct_Quad__Vec3_VA'), void )
    ParamList
      Param( 'self', Quad, l-value )
      Param( 'v', Vec3[], r-value )
    CompoundStatement
      StatementList
        ExprStatement
          AssignOp(ASSIGN_OP)
            StructMemberOp( 'normal' )
              Var('self')
            IndexOp
              Var('v')
              ConstSize( 0 )
  Function( 'entry' ('entry'), void )
    ParamList
    CompoundStatement
      StatementList
        StatementList
          VarDecl( 'v', Vec3[] )
        ExprStatement
          MethodOp( 'push' )
            Var('v')
            ArgList
              TempValue( Vec3 )
                ArgList
                  ConstScalar( 1 )
                  ConstScalar( 2 )
                  ConstScalar( 3 )
        StatementList
          InitializedVarDecl( VarDecl( 'q', Quad ), ArgList )
            ArgList
              Var('v')
        Report
          Var('q')
{normal:{x:1,y:2,z:3}}
